selfdrive/car/gm/carcontroller.py

```diff
@@ -60,20 +71,31 @@ def update(self, CC, CS):
 
       can_sends.append(gmcan.create_steering_control(self.packer_pt, CanBus.POWERTRAIN, apply_steer, idx, CC.latActive))
 
+
+
+    
     if self.CP.openpilotLongitudinalControl:
       # Gas/regen, brakes, and UI commands - all at 25Hz
       if self.frame % 4 == 0:
+        # Pitch compensated acceleration;
+        # TODO: include future pitch (sm['modelDataV2'].orientation.y) to account for long actuator delay
+        pitch = clip(CC.orientationNED[1], self.pitch.x - PITCH_MAX_DELTA, self.pitch.x + PITCH_MAX_DELTA)
+        pitch = clip(pitch, PITCH_MIN, PITCH_MAX)
+        self.pitch.update(pitch)
```

Comment:
Feels like the delta clip and the filter do similar things, any reason for both?